src/world.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    io::Read,
+    path::{Path, PathBuf},
+};
+
+use regex::Regex;
+use serde::Deserialize;
+
+use crate::{Error, ResourceReader};
+
+/// A World is a list of maps files or regex patterns that define a layout of TMX maps.
+/// You can use the loader to further load the maps defined by the world.
+#[derive(Deserialize, PartialEq, Clone, Debug)]
+pub struct World {
+    /// The path first used in a [`ResourceReader`] to load this world.
+    #[serde(skip_deserializing)]
+    pub source: PathBuf,
+    /// The [`WorldMap`]s defined in the world file.
+    pub maps: Option<Vec<WorldMap>>,
+    /// Optional regex pattern to load maps.
+    pub patterns: Option<Vec<WorldPattern>>,
+    /// The type of world, which is arbitrary and set by the user.
+    #[serde(rename = "type")]
+    pub world_type: Option<String>,
+}
+
+/// A WorldMap provides the information for a map in the world and its layout.
+#[derive(Deserialize, PartialEq, Clone, Debug)]
+pub struct WorldMap {
+    /// The filename of the tmx map.
+    #[serde(rename = "fileName")]
+    pub filename: String,
+    /// The x position of the map.
+    pub x: i32,
+    /// The y position of the map.
+    pub y: i32,
+    /// The optional width of the map.
+    pub width: Option<u32>,
+    /// The optional height of the map.
+    pub height: Option<u32>,
+}
+
+/// A WorldPattern defines a regex pattern to automatically determine which maps to load and how to lay them out.
+#[derive(Deserialize, PartialEq, Clone, Debug)]
+pub struct WorldPattern {
+    /// The regex pattern to match against filenames. The first two capture groups should be the x integer and y integer positions.
+    pub regexp: String,
+    /// The multiplier for the x position.
+    #[serde(rename = "multiplierX")]
+    pub multiplier_x: u32,
+    /// The multiplier for the y position.
+    #[serde(rename = "multiplierY")]
+    pub multiplier_y: u32,
+    /// The offset for the x position.
+    #[serde(rename = "offsetX")]
+    pub offset_x: i32,
+    /// The offset for the y position.
+    #[serde(rename = "offsetY")]
+    pub offset_y: i32,
+}
+
+impl WorldPattern {
+    /// Utility function to test a single path against the defined regexp field and returns a parsed WorldMap if it matches.
+    /// Returns none if the filename does not match the pattern.
+    pub fn capture_path(&self, path: &Path) -> Result<WorldMap, Error> {
```

Comment:
I'd suggest to call this `match_path`. In the end it's not the path that is captured, but the path is matched and the map's position is captured.

Reply:
I like match_. I did make another change and ended up with `match_filename` and switched the argument to &str. My thought process is that Tiled doesn't support paths in the regexp field according to the documentation. If we call it match_path it suggests to the user that the field supports paths. While it would technically work fine for rs-tiled if they loaded that file into Tile it would give them issues.

Reply:
Hmm, actually the full paths are matched in Tiled, it's just that only files from the same directory as the world file will be tested on the regex. This means the position can be captured also from somewhere in the path rather than the file name, though I'm not sure that's useful to anybody.

So it doesn't matter much today, but I could imagine a world file could in the future be changed to be able to refer to maps in other directories, so I think `match_path` is in the end the better name.

Reply:
Converted back to match_path and match_paths. match_path takes AsRef<Path> so it's a better interface to work overall.